### PR TITLE
direct people to definitions.html to find out if a field is mutable

### LINKED ISSUE #11141: Document what can be updated with kubectl patch (and how)
The current documentation is very vague https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/kubectl_patch.md and doesn't explain what can be updated in this way. Can I rename or stop the pod, change number of replicas in RC etc.

For example it seems to be impossible to change pod name:
```
kubectl patch pod redis-master-l6lzx -p '{"metadata":{"name":"foobar"}}'
Error from server: the name of the object (foobar) does not match the name on the URL (redis-master-l6lzx)
```

### PR DESCRIPTION
**Pull request by caesarxuchao**
direct people to definitions.html to find out if a field is mutable when doing a kubectl replace/patch;

add an example of `kubectl patch` that involves a merge key.

fixes #11141

@nikhiljindal @mikedanese @bgrant0607

### CODE REVIEW

**Comment by erictune** [comment]
Navigating to the models is just too painful.  Let's not do this until #10966 is done, and then link to that.

**Comment by caesarxuchao** [comment]
I've updated the comment to direct people to check the definitions.html. Could someone review this? Thank you.

File: pkg/kubectl/cmd/replace.go [hunk 1]
```diff
@@ -33,7 +33,9 @@ import (
 const (
 	replace_long = `Replace a resource by filename or stdin.
 
-JSON and YAML formats are accepted.`
+JSON and YAML formats are accepted.
+
+Please refer to the models in http://kubernetes.io/v1.0/docs/api-reference/definitions.html to find if a field is mutable.`
```
Comment (nikhiljindal) [inline pkg/kubectl/cmd/replace.go]: Shouldnt this be linked to HEAD here?
HEAD will be replaced by the appropriate version by versionize-docs.sh

Comment (nikhiljindal) [inline pkg/kubectl/cmd/replace.go]: How about using https://htmlpreview.github.io/?https://github.com/GoogleCloudPlatform/kubernetes/HEAD/docs/api-reference/definitions.html?

File: docs/man/man1/kubectl-patch.1 [hunk 2]
```diff
@@ -18,6 +18,10 @@ Update field(s) of a resource using strategic merge patch
 .PP
 JSON and YAML formats are accepted.
 
+.PP
+Please refer to the models in 
+\[la]https://htmlpreview.github.io/?https://github.com/GoogleCloudPlatform/kubernetes/HEAD/docs/api-reference/definitions.html\[ra] to find if a field is mutable.
+
```
Comment (caesarxuchao) [inline docs/man/man1/kubectl-patch.1]: @nikhiljindal   updated to HEAD address.

**Comment by caesarxuchao** [comment]
@nikhiljindal, sorry for the long delay. I've updated the URL as requested. PTAL. Thank you!

**Comment by nikhiljindal** [comment]
LGTM, thanks!

**Comment by nikhiljindal** [comment]
Looks like you need to run hack/run-gendocs.sh for travis and shippable to pass.

**Comment by caesarxuchao** [comment]
Shippable fails at verify-gendocs.sh, but the PR passes the script on my local machine. Close and reopen to trigger shippable and travis.

**Comment by alex-mohr** [comment]
Travis still complaining "/home/travis/gopath/src/github.com/GoogleCloudPlatform/kubernetes/docs/ is out of date. Please run hack/run-gendocs.sh"

**Comment by alex-mohr** [comment]
LGTM was before last commit, removing LGTM

**Comment by alex-mohr** [comment]
LGTM was before last commit, removing LGTM

**Comment by alex-mohr** [comment]
@k8s-bot test this [contrib/submit-queue: candidate for merging]

**Comment by alex-mohr** [comment]
Automatic merge from SubmitQueue

**Pull request merged**